dftimewolf/cli/dftimewolf_recipes.py

```diff
@@ -68,92 +55,182 @@
+    """Determines the data files path."""
+    data_files_path = os.path.realpath(__file__)
+    data_files_path = os.path.dirname(data_files_path)
+    data_files_path = os.path.dirname(data_files_path)
+    data_files_path = os.path.dirname(data_files_path)
+    data_files_path = os.path.join(data_files_path, 'data')
+
+    if not os.path.isdir(data_files_path):
+      data_files_path = self._DEFAULT_DATA_FILES_PATH
+
+    self._data_files_path = data_files_path
+
+  def _GenerateHelpText(self):
+    """Generates help text with alphabetically sorted recipes.
 
-  parser = argparse.ArgumentParser(
-      formatter_class=argparse.RawDescriptionHelpFormatter,
-      description=generate_help())
+    Returns:
+      str: help text.
+    """
+    recipes = self._recipes_manager.GetRecipes()
+    if not recipes:
+      help_text = '\nNo recipes found.'
+    else:
+      help_text = '\nAvailable recipes:\n\n'
+      for recipe in recipes:
+        short_description = recipe.contents.get(
+            'short_description', 'No description')
+        help_text += ' {0:<35s}{1:s}\n'.format(recipe.name, short_description)
 
-  subparsers = parser.add_subparsers()
+    return help_text
 
-  for recipe in config.Config.get_registered_recipes():
-    subparser = subparsers.add_parser(
-        recipe.name, formatter_class=utils.DFTimewolfFormatterClass,
-        description='{0:s}'.format(recipe.description))
-    subparser.set_defaults(recipe=recipe.contents)
+  def LoadConfiguration(self):
+    """Loads the configuration."""
+    configuration_file_path = os.path.join(self._data_files_path, 'config.json')
+    # Try to open config.json and load configuration data from it.
+    config.Config.load_extra(configuration_file_path)
 
-    for switch, help_text, default in recipe.args:
-      subparser.add_argument(switch, help=help_text, default=default)
-    # Override recipe defaults with those specified in Config
-    # so that they can in turn be overridden in the commandline
-    subparser.set_defaults(**config.Config.get_extra())
+    user_directory = os.path.expanduser('~')
+    configuration_file_path = os.path.join(user_directory, '.dftimewolfrc')
+    config.Config.load_extra(configuration_file_path)
 
-  args = parser.parse_args()
-  recipe = args.recipe
+    configuration_file_path = os.path.join('/', 'etc', 'dftimewolf.conf')
+    config.Config.load_extra(configuration_file_path)
 
-  state = DFTimewolfState(config.Config)
-  print('Loading recipes...')
-  state.load_recipe(recipe)
-  print('Loaded recipe {0:s} with {1:d} modules'.format(
-      recipe['name'], len(recipe['modules'])))
+    configuration_file_path = os.path.join(
+        '/', 'usr', 'share', 'dftimewolf', 'dftimewolf.conf')
+    config.Config.load_extra(configuration_file_path)
 
-  print('Setting up modules...')
-  state.setup_modules(args)
-  print('Modules successfully set up!')
+  def ParseArguments(self, arguments):
+    """Parses the command line arguments.
```

Comment:
This function does more than just parse arguments, namely calls `setup_modules` which already calls logic from each individual module. Maybe rename or extract the `setup_module` call?

Reply:
per conversation, I've changed the function.